stacks-signer/src/signer.rs

```diff
@@ -1609,15 +1706,22 @@ fn load_encrypted_signer_state<S: SignerStateStorage>(
     storage: S,
     id: S::IdType,
     private_key: &Scalar,
-) -> Result<Option<v2::Signer>, PersistenceError> {
-    if let Some(encrypted_state) = storage.get_encrypted_signer_state(id)? {
-        let serialized_state = decrypt(private_key, &encrypted_state)?;
-        let state = serde_json::from_slice(&serialized_state)
-            .expect("Failed to deserialize decryoted state");
-        Ok(Some(v2::Signer::load(&state)))
-    } else {
-        Ok(None)
-    }
+) -> Result<StoredSignerStates, PersistenceError> {
```

Comment:
I am aware that this isn't really new to this PR, but the `signer.rs` file is getting pretty bloated, and these storage routines seem like perfect candidates for separating out into a separate module. How much of a refactor would that require?

Reply:
Factored out to a `storage` module. Let me know if you have any opinions on how this module looks.